### PR TITLE
Remove no-longer-used variables mynp, np0, maxn from collimation

### PR DESCRIPTION
**Pull request by kyrsjo**

### CODE REVIEW

**Comment by amereghe** [comment]
yes - just back to office.
Should do that later in the afternoon

File: source/collimation.f90 [hunk 1]
```diff
@@ -1245,40 +1238,39 @@ subroutine collimate_init()
...
     end if
   end if
+
 !++  Reset distribution for pencil beam
 !
   if(ipencil.gt.0) then
     write(lout,"(a)") "COLL> WARNING Distributions reset to pencil beam!"
     write(outlun,*) 'WARN>  Distributions reset to pencil beam!'
-    do j = 1, mynp
+    do j = 1, napx
```
[the remark this answers is not in the document]
Comment (kyrsjo) [reply source/collimation.f90]: No, this is intentional - it loops over the actual particles (napx, previously mynp), not the memory range (npart, previously maxn).

File: source/collimation.f90 [hunk 2]
```diff
@@ -1297,22 +1289,22 @@ subroutine collimate_init()
...
-    call h5_writeData(setDist0, 6, mynp, myp(1:mynp))
+    call h5_createDataSet("dist0", h5_collID, fmtDist0, setDist0, napx)
+    call h5_prepareWrite(setDist0, napx)
+    call h5_writeData(setDist0, 1, napx, myx(1:napx))
+    call h5_writeData(setDist0, 2, napx, myxp(1:napx))
+    call h5_writeData(setDist0, 3, napx, myy(1:napx))
+    call h5_writeData(setDist0, 4, napx, myyp(1:napx))
+    call h5_writeData(setDist0, 5, napx, mys(1:napx))
+    call h5_writeData(setDist0, 6, napx, myp(1:napx))
     call h5_finaliseWrite(setDist0)
```
[the remark this answers is not in the document]
Comment (kyrsjo) [reply source/collimation.f90]: See above. This is for writing data to file, and we don't want to write junk to the file. Same applies to the writeout to `dist0.dat` (below).

File: source/collimation.f90 [hunk 3]
```diff
@@ -1297,22 +1289,22 @@ subroutine collimate_init()
...
     deallocate(fldDist0)
   else
 #endif
     call funit_requestUnit('dist0.dat', dist0_unit)
     open(unit=dist0_unit,file='dist0.dat') !was 52
     if(dowrite_dist) then
-      do j = 1, mynp
+      do j = 1, napx
```
[the remark this answers is not in the document]
Comment (kyrsjo) [reply source/collimation.f90]: Intentional.

**Comment by kyrsjo** [comment]
Hello, I've updated to the newest master, fixed the things that needed fixing in beamgas, and explained why the way it was done is actually correct.

Very sorry for taking forever to respond after bugging Alessio to do the review!

File: source/collimation.f90 [hunk 4]
```diff
@@ -39,7 +39,7 @@ module collimation
 
 !+cd collpara
   integer, parameter :: max_ncoll  = 100
-  integer, parameter :: maxn       = 20000
+  !integer, parameter :: maxn       = 20000
```
[the remark this answers is not in the document]
Comment (amereghe) [reply source/collimation.f90]: great, thanks!

File: source/collimation.f90 [hunk 5]
```diff
@@ -383,7 +383,6 @@ module collimation
 
 ! IN "+CD DBTRTHIN", "+CD DBDATEN", "+CD DBTHIN6D", and "+CD DBMKDIST"
 ! USED IN MULTIPLE COMMON BLOCKS
-  integer, save :: mynp
   logical, save :: cut_input
```
[the remark this answers is not in the document]
Comment (amereghe) [reply source/collimation.f90]: good!

File: source/collimation.f90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (amereghe) [reply source/collimation.f90]: ok!

**Comment by kyrsjo** [comment]
Thanks!

**Pull request merged**